uptane-standard.md

```diff
@@ -240,28 +330,220 @@ The Time Server SHALL expose a public interface allowing primaries to communicat
+1. Send metadata to secondaries ({{send_metadata_primary}})
+1. Send images to secondaries ({{send_images_primary}})
+
+#### Construct and send vehicle version manifest {#construct_manifest_primary}
+
+The primary SHALL build a *vehicle version manifest* as described in {{vehicle_version_manifest}}.
+
+Once it has the complete manifest built, it MAY send the manifest to the director repository. However, it is not strictly required that the primary send the manifest until step three.
+
+Secondaries MAY send their version report at any time, so that it is stored on the primary already when it wishes to check for updates. Alternatively, the primary MAY request a version report from each secondary at the time of the update check.
+
+#### Download and check current time {#check_time_primary}
+
+The primary SHALL download the current time from the time server, for distribution to its secondaries.
+
+The version report from each secondary ECU (as described in {{version_report}}) contains a nonce, plus a signed ECU version report. The primary SHALL gather each of these nonces from the secondary ECUs, then send them to the time server to fetch the current time. The time server responds as described in {{time_server}}, providing a cryptographic attestation of the last known time. The primary SHALL verify that the signatures are valid, and that the time the server attests is greater than the previous attested time.
+
+#### Download and verify metadata {#download_meta_primary}
+
+The primary SHALL download metadata for all targets and perform a full verification on it as specified in {{full_verification}}.
+
+#### Download and verify images {#download_images_primary}
+
+The primary SHALL download and verify images for itself and for all of its associated secondaries. Images SHALL be verified by checking that the hash of the image file matches the hash specified in the director's targets metadata for that image.
 
-#### Download and check current time {#check_time}
+There may be several different filenames that all refer to the same image binary, as described in {{targets_meta}}. The primary SHALL associate each image binary with each of its possible filenames.
 
-#### Download and verify metadata {#download_meta}
+#### Send latest time to secondaries {#send_time_primary}
 
-#### Download and verify images
+The primary SHALL send the time server's latest attested time to each ECU. The secondary SHALL verify the time message, then overwrite its current time with the received time.
 
-#### Send latest time to secondaries
+#### Send metadata to secondaries {#send_metadata_primary}
 
-#### Send metadata to secondaries
+The primary SHALL send the latest metadata it has downloaded to all of its associated secondaries.
 
-#### Send images to secondaries
+Full verification secondaries SHALL keep a complete copy of all metadata. A partial verification secondary MAY keep *only* the targets metadata file from the director repository.
```

Comment:
@mnm678 Is time server metadata needed?

Reply:
@JustinCappos The secondaries get the latest time from the primary a couple of steps before this, so I don't think they need the time server metadata.

Reply:
I'll change it to "repository metadata".